cogs/poetry.py

```diff
@@ -0,0 +1,85 @@
+from os import path
+from pickle import load
+from typing import List, Tuple
+from random import choice
+from discord.ext import commands
+from .utils.poetry_toolz import PoetryGen, parse_poem_config
+
+
+class Poems(commands.Cog):
+    def __init__(self, bot):
+        self.bot = bot
+        rel_path = path.abspath(path.dirname(__file__))
+        markov_file = open(path.join(rel_path, "utils/rev_markov"), "rb")
+        rhyme_file = open(path.join(rel_path, "utils/rhyme_dict"), "rb")
+        self.poem_machine = PoetryGen(load(markov_file), load(rhyme_file))
+        markov_file.close()
+        rhyme_file.close()
+
+    @commands.command()
+    async def poem(self, ctx, *, command: str = None):
+        """
+		Generate a poem from one of our prebuilt configurations using our (not) state of the poetry generator!
+		Either call "?poem" for a random configration, or call "?poem x" where x is one of the valid prebuilts.
+		Valid prebuilts are: haiku, limerick and alexandrine
+        """
+        configs: List[List[Tuple[str,
+                                 int]]] = [[("A", 5), ("B", 7), ("C", 5)],
+                                           [("A", 10), ("A", 10), ("B", 7),
+                                            ("B", 7), ("A", 10)],
+                                           [("A", 12), ("A", 12), ("A", 12),
+                                            ("A", 12)]]
+        if command is None:
+            await ctx.trigger_typing()
+            await ctx.send("\n".join(self.poem_machine.mk_poem(
+                choice(configs))))
+            return
+        allowed_configs: List[str] = ["haiku", "limerick", "alexandrine"]
+        if command not in allowed_configs:
+            await ctx.send(
+                f"invalid option, valid options are: {allowed_configs}")
+            return
+        await ctx.trigger_typing()
+        if command == "haiku":
+            await ctx.send("\n".join(self.poem_machine.mk_poem(configs[0])))
+        elif command == "limerick":
+            await ctx.send("\n".join(self.poem_machine.mk_poem(configs[1])))
+        elif command == "alexandrine":
+            await ctx.send("\n".join(self.poem_machine.mk_poem(configs[2])))
```

Comment:
this whole if could be simplified to
```python
await ctx.send("\n".join(self.poem_machine.mk_poem(configs[allowed_configs.index(command)])))
```

Reply:
and in fact, configs appear to have names, so `configs` should probably be a dict